packages/core/src/deal-tracker/spade-oracle-sync-tick.js

```diff
@@ -210,3 +215,16 @@ async function getSpadeOracleCurrentState (spadeOracleUrl) {
     ok: dealMap
   }
 }
+
+/**
+ * @param {import('@web3-storage/data-segment').LegacyPieceLink} link
+ * @param {number} height
+ */
+export function convertPieceCidV1toPieceCidV2 (link, height) {
+  const piece = Piece.fromInfo({
+    link,
+    size: Piece.Size.fromHeight(height)
+  })
```

Comment:
You already have a height and root is `link.multihash.digest` so you could alse use [`Piece.view`](https://github.com/web3-storage/data-segment/blob/main/src/piece.js#L77-L81) and skip whole info creation piece, that said you would have to do something like

```suggestion
  const piece = Piece.fromInfo({
    root: link.multihash.digest,
    height,
    padding: 0n
  })
```

If we do know original CAR size however it would be great to compute padding also instead of just putting `0n` however.

Reply:
I think we need the size here and not height: https://github.com/web3-storage/data-segment/blob/main/src/piece.js#L104

This is the aggregate piece CID that Spade Oracle has, while tracking Deals (uses the old Piece Cid v1 + height)

Reply:
> I think we need the size here and not height: https://github.com/web3-storage/data-segment/blob/main/src/piece.js#L104

Sorry I had typo I meant `Piece.toView` function in the suggestion, which takes `{height, root, padding}` triple.

Reply:
> This is the aggregate piece CID that Spade Oracle has, while tracking Deals (uses the old Piece Cid v1 + height)

Ah ok, aggregates don't really have padding so that makes sense. I actually wanted to use different CIDs for aggregates, but also don't really want to argue for them in the spec.

Reply:
cool thanks, changed to suggestion